### DISCUSSION TITLE
Can't open the reference design in FlashPro Express

### DISCUSSION THREAD

**Discussion by sand7000** [polarfire-soc/polarfire-soc-documentation#308] [Reference designs, FPGA and development kits] [answered]
I downloaded the reference design here:

https://github.com/polarfire-soc/icicle-kit-reference-design/releases/download/v2023.02/MPFS_ICICLE_BASE_DESIGN_2023_02.zip

and I want to install it on my new polarfire icicle board.

I installed FlashPro Express on both Linux and Windows 10. Both give the same error when I try to open the job file:

```
Error: Failed to open the project file '/home/me/projects/icicle/mpfs_base/MPFS_ICICLE_BASE_DESIGN_2023_02/MPFS_ICICLE_KIT_BASE_DESIGN_2023_02.job'.
```

That's the only error or message I get which isn't a lot to go on. I watched this video:

https://www.youtube.com/watch?v=BY3306ITDhM&list=PL9B4edd-p2ahc3AQdBAmGfAyo_pcDW0Tg&index=4

which mentioned that older builds might not have the USB embedded Programming connector enabled but when I power the board up I can see that linux dmesg indicates that it recognizes the connected device:

```
[16679.104542] usb 2-4.1.1.2: USB disconnect, device number 19
[16733.608094] usb 2-4.1.1.2: new high-speed USB device number 20 using xhci_hcd
[16733.708697] usb 2-4.1.1.2: New USB device found, idVendor=1514, idProduct=200b, bcdDevice=ef.9b
[16733.708701] usb 2-4.1.1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[16733.708701] usb 2-4.1.1.2: Product: Embedded FP6b
[16733.708702] usb 2-4.1.1.2: Manufacturer: Microchip
[16733.708703] usb 2-4.1.1.2: SerialNumber: 04FD11E1
[16733.711814] hid-generic 0003:1514:200B.000E: hiddev97,hidraw5: USB HID v1.10 Device [Microchip Embedded FP6b] on usb-0000:00:14.0-4.1.1.2/input0
```
and windows indicated it has discovered and set up the device as well. 

The only odd thing is that when I click "Open Job Project" the only extension options are "all files" and ".pro". The user guide indicates that FlashPro Express uses .job files instead of the older .pro files used by FlashPro. Maybe it just hasn't been revised to default to the newer file extension? Maybe I need a different version of FlashPro or FlashPro express than what comes with the newest Libero?

Thoughts/suggestions?

**Comment by andrenemat** [comment, marked as answer]
With FlashPro Express, you need to create a new programming project, and import the *.job file into this project. When you open FlashPro Express, click on "New", or Project -> New Project, then in the menu that opens up, on the first line import your job file from the *.zip file you have downloaded, then tell the tool where you want it to store the created Job Project. Next time if you want to program the same FPGA image, you can "Open" a job project and this time you will see the .pro file in the folder where you had created the job project above.

If you have the Libero Suite installed, you will already have FlashPro Express, please make sure you have the latest version 2022.3. If not please make sure you have the latest version of the Programming and Debugging tools from:
https://www.microchip.com/en-us/products/fpgas-and-plds/fpga-and-soc-design-tools/programming-and-debug#Download%20Software

**Reply by sand7000** [reply]
This cleared up my confusion thanks!
